File: src/Domain/Entities/Games/RoundSummary.cs
using Domain.Entities.Common;
using Domain.Enums;

namespace Domain.Entities.Games;

public class RoundSummary
{
    public RoundSummaryStatus Status { get; set; }
    public string UserId { get; set; }
    public int Score { get; set; }
    public int TimePassed { get; set; }
    public bool SolutionShared { get; set; }
    public Solution Solution { get; set; }
    public List<TestSummary> TestSummaries { get; set; }
}

File: src/react-frontend/src/app/types/models/game/roundSummary.ts
import { RoundSummaryStatus } from 'app/types/enums/roundSummaryStatus';
import { Solution } from '../general/solution';
import { UserDto } from '../user/userDto';
import { TestSummary } from './testSummary';

export type RoundSummary = {
  status: RoundSummaryStatus;
  user: UserDto;
  score: number;
  timePassed: number;
  solutionShared: boolean;
  solution?: Solution;
  testSummaries: TestSummary[];
};
